patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -1116,6 +1194,21 @@
 +    }
 +
 +    /**
++     * Common way to recolor a block with an external tool
++     * @param world The world
++     * @param pos Block position in world
++     * @param side The side hit with the coloring tool
++     * @param color The color to change to
++     * @param player The player that's attempting to recolor the block
++     * @param hit The position on the block where the player clicked
++     * @return If the recoloring was successful
++     */
++    public boolean recolorBlock(World world, BlockPos pos, EnumFacing side, net.minecraft.item.EnumDyeColor color, EntityPlayer player, RayTraceResult hit)
++    {
```

Comment:
Just like in the case of rotation, the old recolorBlock() is more generic and would affect the whole block, and the new one would affect only a specific part of it (if coded to do so). Also, it's nice to know the player that's trying to recolor it. I think it should be supplied whenever possible to avoid people trying to get it from the raytrace and doing messy things like that. As long as it's not completely unnecessary information (like https://xkcd.com/221/), it's always to know more information about an event. Who knows when it may come in handy!